src/battle_ai_switch_items.c

```diff
@@ -208,28 +231,81 @@ static bool32 HasBadOdds(u32 battler, bool32 emitResult)
 
             // Switch mon out
             gBattleStruct->AI_monToSwitchIntoId[battler] = PARTY_SIZE;
-            if (emitResult)
-                BtlController_EmitTwoReturnValues(battler, 1, B_ACTION_SWITCH, 0);
-            return TRUE;
+            if (IsSwitchinIdValid(battler, gBattleStruct->AI_monToSwitchIntoId[battler]))
+            {
+                if (emitResult)
+                    BtlController_EmitTwoReturnValues(battler, 1, B_ACTION_SWITCH, 0);
+                return TRUE;
+            }
         }
     }
     return FALSE;
 }
 
 static bool32 ShouldSwitchIfAllBadMoves(u32 battler, bool32 emitResult)
 {
-    if (AI_DATA->shouldSwitchIfBadMoves & (1u << battler))
+    u32 i, j;
```

Comment:
Could you use more descriptive loop variables? Especially if more several loops are inside one function it increases readability greatly. Maybe for i `moveIndex` and for j `battlerDef`?

Reply:
Great idea, I was just copy pasting from Egg's version. Pushed :)